platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,74 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects. Create a module that will provide simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+There are two issues that need to be addressed:
+
+* We need to have a public API for the Merkle tree that is simple and easy to use, residing in a separate module with a minimal set of dependencies.
+* The Merkle tree has to be fully expressed in terms of State API. Currently, `PlatformState` stands out and needs to be refactored.
+
+The goal of this design proposal is to ensure that one can interact with all the objects in the Merkle tree using the newly introduced State API.
+
+This refactoring will enable optimized state proof, as it will be possible to represent an n-ary root node as a virtual binary tree.
+Therefore, the number of hashes in the state proof will be minimized.
+
+## Requirements
+
+- `PlatformState` should be refactored to a singleton object as defined by the State API.
+- the Merkle tree should be reshaped so that `MerkleHederaState` becomes the root of the tree and `PlatformState` becomes its child.
```

Comment:
These two items are implementation details, not requirements. I think the requirement is for every part of the state, including `PlatformState` to be accessible through the State API.

The second bullet is unclear. Why is this being specified? What does this change mean to the users of the State API?

Reply:
I removed this part out of the proposal